src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3260,18 +3294,41 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
 
                     if (real)
                     {
-                        double d;
-                        if (Double.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out d))
+                        if (Double.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out double d))
                         {
                             // TryParse incorrectly return +0 when the result should be -0, so check for that case
                             if (d == 0.0 && strNum[0] == '-')
                             {
                                 d = -0.0;
                             }
-                            if (suffix == 'l' || suffix == 'L')
+
+                            if (suffix == NumberSuffixFlags.Long)
                             {
                                 result = ((long)Convert.ChangeType(d, typeof(long), CultureInfo.InvariantCulture)) * multiplier;
                             }
+
+                            if (suffix == NumberSuffixFlags.Short)
+                            {
+                                result = (short)((short)Convert.ChangeType(d, typeof(short), CultureInfo.InvariantCulture) * multiplier);
```

Comment:
I'll see how it behaves with that conversion in the tokenizer. It's been my experience so far that Convert.ChangeType() seems to do Banker's Rounding whereas casting and some other methods... do different things.

Reply:
That's probably also functional. But as far as I can tell that offers no noticeable advantage over `Convert.ToInt16(d * multiplier)` which is what I've gone with at the present.

Reply:
Just to go on the record -- since we're looking for specific rounding behaviour here, my preference would be to be explicit about that rather than getting it from casts